### PR TITLE
feat(forge): add calldata and returndata views to debugger

### LINKED ISSUE #5492: feat(forge): Forge debugger should have calldata and returndata buffer views
### Component

Forge

### Describe the feature you would like

The Forge debugger should be able to view both the calldata and returndata buffers. Currently, only the stack and memory are viewable.
Visually, this could be achieved by adding the ability to toggle or cycle the standard panels to returndata/calldata views.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by emo-eth**
## Motivation

It's often useful to see what data is actually in calldata/returndata, as well as see which regions are being operated on.

Closes #5492.

## Solution

- Add calldata and returndata to `DebugStep`
- Add `active_buffer` to `DebuggerContext`
- `get_memory_access` -> `get_buffer_accesses`; note which buffer is being read from, if any; simplify return type
- `draw_memory` -> `draw_buffer`; compare read/writes to current active buffer 
- add `b` ("buffer") operator to debugger to cycle Memory -> Calldata -> Returndata buffers in the "buffer" panel view
- add `b` to help string; update references of "memory" to "buffer"

Unsure of where debugger tests live (if anywhere), but draw logic seems to handle edge cases well; I'd be most concerned about potential performance issues or highlighting regions in the wrong buffer.

Screenshots:

Calldata access:
Screenshot 2024-02-13 at 2 10 15 PM
Memory overwrite from calldatacopy:
Screenshot 2024-02-13 at 2 11 32 PM
Returndata access:
Screenshot 2024-02-13 at 2 10 42 PM
Memory overwrite from returndatacopy:
Screenshot 2024-02-13 at 2 10 55 PM

### CODE REVIEW

File: crates/debugger/src/tui/context.rs [hunk 1]
```diff
@@ -13,10 +13,58 @@ pub(crate) struct DrawMemory {
...
+pub(crate) enum ActiveBuffer {
+    #[default]
+    Memory,
+    Calldata,
+    Returndata,
+}
+
+/// Used to note which buffer is being read from by an opcode, if any.
+#[derive(Debug, Default)]
+pub(crate) enum BufferReadAccess {
```
Comment (DaniPopes) [inline crates/debugger/src/tui/context.rs:31]: We could just use `Option<ActiveBuffer>`

**Review by DaniPopes — CHANGES_REQUESTED** [review]
Nice, I like this, just a nit

**Review by mattsse — APPROVED** [review]
sweet!
pending @DaniPopes

**Review by Evalir — APPROVED** [review]
awesome work. Pending clippy & @DaniPopes
